solver/src/solver/single_order_solver.rs

```diff
@@ -1,21 +1,55 @@
 use std::time::Instant;
 
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use futures::future;
 use rand::prelude::SliceRandom;
 
 use crate::{
     liquidity::{LimitOrder, Liquidity},
     settlement::Settlement,
-    solver::Solver,
+    solver::{solver_utils::SettlementError, Solver},
```

Comment:
Not directly relevant to this PR but I think this type SettlementError should move into this file because it is part of the single order solver abstraction.